model-optimizer/mo/middle/passes/mean_scale_values.py

```diff
@@ -20,62 +20,130 @@
 from mo.middle.pattern_match import apply_pattern
 
 
-def move_scaleshift_to_preprocess_action(graph, match):
+def move_mean_values_to_preprocess_action(graph, match):
     mean_values = {}
+    if 'const_mul_output' in match and match['const_mul_output'].has('value'):
+        weights = match['const_mul_output'].value
+        if any([x != 1 for x in weights]):
+            return
+
+    # # Keep biases (mean values) for current input as graph attr and remove mean values layers
+    # # Input->data->Mul->Add->scsh_data    =>    Input->scsh_data
+    # # Input->data->Mul->scsh_data    =>    Input->scsh_data
+    # # Input->data->Add->scsh_data    =>    Input->scsh_data
     input_op = match['input_op']
-    scale_shift = match['scale_shift']
-    weights = np.squeeze(match['weights'].value)
-    biases = np.squeeze(match['biases'].value)
 
-    if graph.graph['cmd_params'].reverse_input_channels:
-        biases = np.flip(biases)
-
-    if any([x != 1 for x in weights]):
-        return
-
-    # Keep biases (mean values) for current input as graph attr and remove ScaleShift layer
-    # Input->data->ScaleShift->scsh_data    =>    Input->scsh_data
-    graph.remove_edge(input_op.id, input_op.out_node().id)
-    graph.add_edge(input_op.id, scale_shift.out_node().id, out=0)
-    graph.remove_edge(scale_shift.id, scale_shift.out_node().id)
-
-    # If bias contains zeros we just remove it
-    if all([x == 0 for x in biases]):
-        return
+    if 'add' in match:
+        add_outputs = list(match['add'].out_ports().values())
```

Comment:
there is no need to check number of out ports of `match['add']`. It always exist. It may be not connected to anywhere but the function `match['add'].out_port(0).get_connection().set_source(...)` will still work. So, just remove lines 37 and 38